### ISSUE TITLE
Claims Detail (Active/Closed): Content/UI updates within the Files Segment

### ISSUE THREAD

**Issue by wavelaurenrussell** [department-of-veterans-affairs/va-mobile-app#8942]
### Description
This story will cover content and UI updates within the Files Segment on all Claim types. This list will display for all claims with uploaded media, whether the status is Active or Closed.

### ACs
- [ ] Use a row component to represent each file associated with claim
- [ ] Within the row component, display these additional fields associated with each file: 

    - Filename
    - Request type
    - Document type
    - Received

- [ ] VQA Misty / Lauren

#### Resources
- Copy doc: [fields within each file](https://docs.google.com/spreadsheets/d/16waC665d7uNRQmM83SJIm9bTDyRePOuB6r9G_aQE8Tw/edit?gid=789133900#gid=789133900&range=13:13)
- Figma mockup: [list design
](https://www.figma.com/design/kSlXLwTVLunM0fqCG95k5x/%F0%9F%94%8D-Claims-2.0---Working---VA-Mobile?node-id=1587-7101&t=hmtlgXbIWLyCFFic-4)

**Comment by rbontrager** [comment]
Still waiting on one more content change.  Moving the ticket back to in progress until we determine if any changes need to be done.

**Comment by rbontrager** [comment]
After adding files to the file request none of the files are appearing in the file segment.  Dylan thinks he has found a cause so sending the ticket back.

**Comment by rbontrager** [comment]
After adding files to the file request none of the files are appearing in the file segment.  Dylan thinks he has found a cause so sending the ticket back.

**Comment by rbontrager** [comment]
@Sparowhawk I'm now seeing the files in the files segment, yay. A couple things:

1. From the thread during VQA: Misty asked that the received field be in Month Day, Year format.  I'm not seeing that reflected in the list.
2. Is it possible to add a refresh whenever the user opens the file segment? Currently if you finish a file request and navigate to the files segment, the new files aren't appearing in the list.

**Comment by Sparowhawk** [comment]
Updated the date to the correct format

**Comment by Sparowhawk** [comment]
Added a refresh when the files segment reappears

**Comment by rbontrager** [comment]
Verified the following:
- The files segment replaces the details segment
- Each file for the claim appears for both open and closed claims and matches the AC above.  I went through as many claims as I could in Charles Proxy and verified that each file in the json was represented in the files segment.
- If there are no files, then 'This claim does not have any files yet' appears in the files segment for both open and closed claims
- Each files received field is displayed in Month Day, Year format and matches the rest of the date formats in claims
- Each document type and file type is displayed properly in the file document type field
- Multiple files in one file request are displayed in the files segment
- A11y smoke tests

Approved by QA.
